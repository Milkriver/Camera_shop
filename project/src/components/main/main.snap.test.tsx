import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import HistoryRouter from '../history-router/history-router';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import Main from './main';

const history = createMemoryHistory();

describe('Component: Main', () => {
  it('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
      DATA: {
        activePaginationPage: 1,
      },
      OFFERS: {
        offers: []
      }
    });
    const {container} = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <Main/>
        </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
