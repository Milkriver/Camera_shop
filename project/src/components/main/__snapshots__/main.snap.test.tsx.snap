// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Main should render correctly 1`] = `
<div>
  <main>
    <div
      class="banner"
    />
    <div
      class="page-content"
    >
      <div
        class="breadcrumbs"
      >
        <div
          class="container"
        >
          <ul
            class="breadcrumbs__list"
          >
            <li
              class="breadcrumbs__item"
            >
              <a
                class="breadcrumbs__link"
                href="/"
              >
                Главная
                <svg
                  aria-hidden="true"
                  height="8"
                  width="5"
                >
                  <use
                    xlink:href="#icon-arrow-mini"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <section
        class="catalog"
      >
        <div
          class="container"
        >
          <h1
            class="title title--h2"
          >
            Каталог фото- и видеотехники
          </h1>
          <div
            class="page-content__columns"
          >
            <div
              class="catalog__aside"
            >
              <div
                class="catalog-filter"
              >
                <form
                  action="#"
                >
                  <h2
                    class="visually-hidden"
                  >
                    Фильтр
                  </h2>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Цена, ₽
                    </legend>
                    <div
                      class="catalog-filter__price-range"
                    >
                      <div
                        class="custom-input"
                      >
                        <label>
                          <input
                            name="price"
                            placeholder="от"
                            type="number"
                          />
                        </label>
                      </div>
                      <div
                        class="custom-input"
                      >
                        <label>
                          <input
                            name="priceUp"
                            placeholder="до"
                            type="number"
                          />
                        </label>
                      </div>
                    </div>
                  </fieldset>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Категория
                    </legend>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          checked=""
                          name="photocamera"
                          readonly=""
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Фотокамера
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="videocamera"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Видеокамера
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Тип камеры
                    </legend>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          checked=""
                          name="digital"
                          readonly=""
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Цифровая
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          disabled=""
                          name="film"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Плёночная
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="snapshot"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Моментальная
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          checked=""
                          disabled=""
                          name="collection"
                          readonly=""
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Коллекционная
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Уровень
                    </legend>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          checked=""
                          name="zero"
                          readonly=""
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Нулевой
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="non-professional"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Любительский
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="professional"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Профессиональный
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <button
                    class="btn catalog-filter__reset-btn"
                    type="reset"
                  >
                    Сбросить фильтры
                  </button>
                </form>
              </div>
            </div>
            <div
              class="catalog__content"
            >
              <div
                class="catalog-sort"
              >
                <form
                  action="#"
                >
                  <div
                    class="catalog-sort__inner"
                  >
                    <p
                      class="title title--h5"
                    >
                      Сортировать:
                    </p>
                    <div
                      class="catalog-sort__type"
                    >
                      <div
                        class="catalog-sort__btn-text"
                      >
                        <input
                          checked=""
                          id="sortPrice"
                          name="sort"
                          readonly=""
                          type="radio"
                        />
                        <label
                          for="sortPrice"
                        >
                          по цене
                        </label>
                      </div>
                      <div
                        class="catalog-sort__btn-text"
                      >
                        <input
                          id="sortPopular"
                          name="sort"
                          type="radio"
                        />
                        <label
                          for="sortPopular"
                        >
                          по популярности
                        </label>
                      </div>
                    </div>
                    <div
                      class="catalog-sort__order"
                    >
                      <div
                        class="catalog-sort__btn catalog-sort__btn--up"
                      >
                        <input
                          aria-label="По возрастанию"
                          checked=""
                          id="up"
                          name="sort-icon"
                          readonly=""
                          type="radio"
                        />
                        <label
                          for="up"
                        >
                          <svg
                            aria-hidden="true"
                            height="14"
                            width="16"
                          >
                            <use
                              xlink:href="#icon-sort"
                            />
                          </svg>
                        </label>
                      </div>
                      <div
                        class="catalog-sort__btn catalog-sort__btn--down"
                      >
                        <input
                          aria-label="По убыванию"
                          id="down"
                          name="sort-icon"
                          readonly=""
                          type="radio"
                        />
                        <label
                          for="down"
                        >
                          <svg
                            aria-hidden="true"
                            height="14"
                            width="16"
                          >
                            <use
                              xlink:href="#icon-sort"
                            />
                          </svg>
                        </label>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="cards catalog__cards"
              />
              <div
                class="pagination"
              >
                <ul
                  class="pagination__list"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </main>
</div>
`;
